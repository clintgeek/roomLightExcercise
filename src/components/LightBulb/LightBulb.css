.lightBulb {
  fill: black;
  margin-bottom: 1rem;
}

.powered {
  fill: yellow;
}
