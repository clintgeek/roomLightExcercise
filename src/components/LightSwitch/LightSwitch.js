import React from 'react';
import Switch from '@material-ui/core/Switch';

function LightSwitch(props) {
  return (
    <div>
      <Switch
        checked={props.powered}
        onChange={props.togglePower}
        name="power"
        inputProps={{ 'aria-label': 'power' }}
      />
    </div>
  );
}

export default LightSwitch;