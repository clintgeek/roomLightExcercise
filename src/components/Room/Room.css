.roomName {
  padding-bottom: 1rem;
}

.walledRoom {
  border: 1px solid darkgray;
  padding: 20px;
  background: #222222;
}

.bright {
  border: 1px solid black;
  background: lightgray;
}