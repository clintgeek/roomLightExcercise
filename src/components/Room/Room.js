import React, { useState } from 'react';
import LightBulb from '../LightBulb/';
import LightSwitch from '../LightSwitch';
import './Room.css';

function Room(props) {
  const [ powered, setPower ] = useState(false);

  const togglePower = () => setPower(!powered);

  var lightLevelClass = powered ? ' bright' : '';
  const roomClass = 'walledRoom' + lightLevelClass;

  return (
    <div>
      <div className="roomName">{props.name}</div>
      <div className={roomClass}>
        <LightBulb powered={powered}/>
        <LightSwitch powered={powered} togglePower={togglePower} />
      </div>
    </div>
  );
}

export default Room;