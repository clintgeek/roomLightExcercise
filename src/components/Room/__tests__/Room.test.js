import React from 'react';
import {shallow} from 'enzyme';
import Room from '../Room';
import LightBulb from '../../LightBulb';
import LightSwitch from '../../LightSwitch';

describe('<Room> component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Room name="test room" />);
  })

  it('should match snapshot', () => {
    expect(wrapper.debug()).toMatchSnapshot();  
  });

  it('should render with the room name provided in props', () => {
    expect(wrapper.find('.roomName').text()).toEqual('test room');
  })

  it('should render a <LightBulb> with expected props', () => {
    const lightBulb = wrapper.find(LightBulb);
    expect(lightBulb.isEmptyRender()).toBe(false);
    expect(lightBulb.props()).toEqual({'powered': false})
  });

  it('should render a <LightSwitch> with expected props', () => {
    const lightSwitch = wrapper.find(LightSwitch);
    expect(lightSwitch.isEmptyRender()).toBe(false)
    expect(lightSwitch.props().powered).toEqual(false);
    expect(lightSwitch.props().togglePower).toBeInstanceOf(Function);
  });

  it('should maintain and modify the lights power in state', () => {
    expect(wrapper.find(LightSwitch).props().powered).toEqual(false);
    wrapper.find(LightSwitch).props().togglePower();
    expect(wrapper.find(LightSwitch).props().powered).toEqual(true);
  });

  it('should devise className from state', () => {
    expect(wrapper.find('.bright').isEmptyRender()).toBe(true);
    wrapper.find(LightSwitch).props().togglePower();
    expect(wrapper.find('.bright').isEmptyRender()).toBe(false);
  });
});