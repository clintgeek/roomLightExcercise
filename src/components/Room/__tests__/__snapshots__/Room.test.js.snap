// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Room> component should match snapshot 1`] = `
"<div>
  <div className=\\"roomName\\">
    test room
  </div>
  <div className=\\"walledRoom\\">
    <LightBulb powered={false} />
    <LightSwitch powered={false} togglePower={[Function: togglePower]} />
  </div>
</div>"
`;
